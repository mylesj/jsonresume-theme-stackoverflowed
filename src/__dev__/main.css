html {
    margin: 0;
    padding: 0;
    font-size: 100%;
}

body {
    margin: 0;
    padding: 1rem;
    background-color: #f5f5f5;
}

#root {
    margin: 0 auto;
    min-width: 800px;
    max-width: 1200px;
    box-shadow: 4px 4px 8px rgba(0, 0, 0, 0.25);
}
