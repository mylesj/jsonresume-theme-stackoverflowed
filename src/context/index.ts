import { createContext, useContext } from 'react'

import { Dependencies, Configuration } from '~/types'
import { THEME_NAME } from '~/constants'

export const AppContext = createContext<Dependencies>({
    resume: {},
})

export const useAppContext = () => useContext(AppContext)

export const useResume = () => useAppContext().resume

export const useConfig = <T extends keyof Configuration>(
    key: T
): Configuration[T] => useResume().meta?.[THEME_NAME]?.[key]
