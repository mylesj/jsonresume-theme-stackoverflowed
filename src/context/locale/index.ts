import { DEFAULT_LOCALE } from '~/constants'

import { countryNameFactory } from './i18nCountries'
import { dateFormatterFactory } from './i18nDates'

export type Locale = unknown &
    Awaited<ReturnType<typeof countryNameFactory>> &
    Awaited<ReturnType<typeof dateFormatterFactory>>

const MEMO: {
    [key in string]: Locale
} = {}

export const getLocale = async (locale = DEFAULT_LOCALE) => {
    if (!(locale in MEMO)) {
        const [countryNameFormatters, dateFormatters] = await Promise.all([
            countryNameFactory(locale),
            dateFormatterFactory(locale),
        ])

        MEMO[locale] = {
            ...countryNameFormatters,
            ...dateFormatters,
        }
    }

    return MEMO[locale]
}
