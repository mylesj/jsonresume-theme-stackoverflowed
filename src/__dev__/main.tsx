import React from 'react'
import ReactDOM from 'react-dom/client'

import { AppContext } from '~/context'
import View from '~/view'

import resume from './sample.resume.json'

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
ReactDOM.createRoot(document.getElementById('root')!).render(
    <React.StrictMode>
        <AppContext.Provider value={{ resume }}>
            <View />
        </AppContext.Provider>
    </React.StrictMode>
)
