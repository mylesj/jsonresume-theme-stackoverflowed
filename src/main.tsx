import { renderToString } from 'react-dom/server'

import { Renderer, RenderOptions } from '~/types'
import { AppContext, getLocale } from '~/context'
import { THEME_NAME } from '~/constants'

import html from './html'
import View from './view'

import { resumeMiddleware } from './middleware'

import pkg from '../package.json'

export const render: Renderer = async (resume) => {
    const enhancedResume = await resumeMiddleware(resume)
    const locale = await getLocale(resume.meta?.[THEME_NAME]?.locale)

    return html({
        resume,
        meta: {
            generator: `${pkg.name}@${pkg.version}`,
        },
        body: renderToString(
            <AppContext.Provider value={{ resume: enhancedResume, locale }}>
                <View />
            </AppContext.Provider>
        ),
    })
}

export const pdfRenderOptions: RenderOptions = {
    mediaType: 'print',
}
