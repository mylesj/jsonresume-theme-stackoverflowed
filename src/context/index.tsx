import { createContext, useContext } from 'react'

import { ResumeSchema, Configuration } from '~/types'
import { THEME_NAME } from '~/constants'

import { Locale } from './locale'

export { getLocale } from './locale'

export type Dependencies = {
    resume: ResumeSchema
    locale: Locale
}

export const AppContext = createContext<Dependencies | null>(null)

const useAppContext = () => useContext(AppContext)

export const useResume = () => useAppContext()?.resume as ResumeSchema

export const useLocale = () => useAppContext()?.locale as Locale

export const useConfig = <T extends keyof Configuration>(
    key: T
): Configuration[T] => useResume()?.meta?.[THEME_NAME]?.[key]
