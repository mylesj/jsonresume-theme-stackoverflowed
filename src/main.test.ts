/**
 * @jest-environment node
 */

import { render } from './main'
import { useResume } from './context'

jest.mock('./view', () => ({
    __esModule: true,
    default: () => useResume().basics?.name,
}))

it('should render a basic html document', async () => {
    const rendered = await render({
        basics: {
            name: 'Richard Hendriks',
            label: 'Programmer',
        },
    })

    expect(rendered).toMatchSnapshot()
})
