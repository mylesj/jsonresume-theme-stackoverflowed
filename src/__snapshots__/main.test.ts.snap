// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a basic html document 1`] = `
"<!DOCTYPE html>
    <html lang=\\"en\\">
        <head>
            <meta charset=\\"utf-8\\" />
            <meta
                name=\\"viewport\\"
                content=\\"width=device-width, initial-scale=1\\"
            />
            <meta name=\\"generator\\" content=\\"jsonresume-theme-stackoverflowed@0.0.0-development\\" />
            <title>
                Richard Hendriks - Programmer - Resume
            </title>
        </head>
        <body>
            <div id=\\"root\\">Richard Hendriks</div>
        </body>
    </html>"
`;
